### PR TITLE
ci: add snyk for container image

### PR DESCRIPTION
**Pull request by Madhu-1**
adding a GitHub action to do security scanning for the cephcsi docker image

### CODE REVIEW

**Comment by Madhu-1** [comment]
Testing PR, will update the PR to run it weekly and on release/tag once testing is done.

**Comment by Madhu-1** [comment]
>For secrets stored at the environment level, you can enable required reviewers to control access to the secrets. A workflow job cannot access environment secrets until approval is granted by required approvers.

@nixpanic @Rakshith-R can someone place approval which is required to access secret in this action?

**Review by nixpanic — DISMISSED** [review]
Approved for testing

**Comment by nixpanic** [comment]
> Approved for testing

Restarted the job, see https://github.com/ceph/ceph-csi/actions/runs/6887739362/job/18736026287

**Comment by Madhu-1** [comment]
Ci job ran but github doesnt allow us to use the secret in the pull request.

**Review by nixpanic — APPROVED** [review]

File: .github/workflows/snyk-container-image.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  contents: read
+
+jobs:
+  snyk:
+    if: github.repository == 'ceph/ceph-csi'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build a Docker image
+        run: make containerized-build
```
Comment (nixpanic) [inline .github/workflows/snyk-container-image.yaml]: Actually, you need to run `make image-cephcsi`

**Review by yati1998 — DISMISSED** [review]

**Review by nixpanic — CHANGES_REQUESTED** [review]

File: .github/workflows/snyk-container-image.yaml (hunk 1 above)
Comment (Madhu-1) [reply .github/workflows/snyk-container-image.yaml]: yes thats correct 🤦🏻 , fixed now

**Review by nixpanic — APPROVED** [review]

**Review by riya-singhal31 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@mergifyio queue

**Pull request merged**